docs/pages/x/api/data-grid/grid-api.json

```diff
@@ -234,6 +251,11 @@
       "isProPlan": true
     },
     "publishEvent": { "type": { "description": "GridEventPublisher" }, "required": true },
+    "queueChildrenFetch": {
```

Comment:
Instead of exposing the `fetchTopLevelRows` and `queueChildrenFetch` methods, we can expose a single method with optional `rowId`.
What do you think?

Also, would `NestedDataManager` be used for row grouping as well? This might be something to consider now because row groups will rely on `groupKey`, not row id.

Reply:
> Instead of exposing the fetchTopLevelRows and queueChildrenFetch methods, we can expose a single method with optional rowId.

Makes sense. Although it would be a little less explicit, but one less API for the users to consider. Done ✅ 

> row groups will rely on groupKey, not row id.

I am honestly not sure about this. As per my current thinking, row grouping (with data source) is more aligned with tree data (with data source) on the client side rather than client-side row grouping where the Data Grid handles all the computation.

Mainly because all the computation around grouping the rows and generating IDs would happen on the back end. The Data Grid would send an extra parameter (`groupFields`) in every `getRows` call and expect rows for a specific level just like for tree data.

Also similar to the tree data, the Data Grid would use `getChildrenCount` to show an expand icon to the parents. For multiple levels (more than 1 grouping criteria), the Data Grid would pass `groupKeys` as `['Level 1', 'Level 2']`, similar to the tree data.

At least for now, I'd expect `GridRowId` to be the unique identifier for a `NestedDataManager` request.

Does it make sense?

Reply:
> Considering this, I'd expect GridRowId to be the unique identifier for a NestedDataManager request.

Makes sense to me. The only difference is that in the case of row grouping, `rowId` is autogenerated.
We already have the `setRowChildrenExpansion` method that accepts `rowId` and works for both tree data and row grouping 👍🏻

Reply:
> The only difference is that in the case of row grouping, rowId is autogenerated.

Correct. We can treat row grouping (with data source) `rowId`s as normal IDs (not auto-generated ones) since they will be auto-generated on the backend. The Data Grid won't care whether or not they are auto-generated as long as they are being passed.